platform/ios/app-swift/Sources/LocationPrivacyExample.swift

```diff
@@ -9,75 +9,114 @@ enum LocationAccuracyState {
     case fullAccuracy
 }
 
-class Coordinator: NSObject, MLNMapViewDelegate {
-    @Binding var mapView: MLNMapView
-    @Binding var locationAccuracy: LocationAccuracyState
-    var pannedToUserLocation = false
-
-    init(mapView: Binding<MLNMapView>, locationAccuracy: Binding<LocationAccuracyState>) {
-        _mapView = mapView
-        _locationAccuracy = locationAccuracy
+class MapViewModel: NSObject, ObservableObject {
+    // Weak because the coordinator has a strong reference to the view model
+    weak var mapCoordinater: MapLibreRepresentableCoordinator?
+    @MainActor @Published var locationAccuracy: LocationAccuracyState = .unknown
+
+    @MainActor func requestTemporaryLocationAuthorization() {
+        print("Requesting precice location")
+
+        switch locationAccuracy {
+        case .reducedAccuracy:
+            let purposeKey = "MLNAccuracyAuthorizationDescription"
+            mapCoordinater?.mapView?.locationManager.requestTemporaryFullAccuracyAuthorization!(withPurposeKey: purposeKey)
```

Comment:
Force unwrap is usually a bad idea. Also, should we really be using the locationManager within mapView and not one that the app initializes for its own purposes?

Reply:
Great catch, fixed.